test/integration/importer/importer_test.go

```diff
@@ -0,0 +1,155 @@
+
+		flavor = utiltesting.MakeResourceFlavor("f1").Obj()
+		gomega.Expect(k8sClient.Create(ctx, flavor)).To(gomega.Succeed())
+
+		cq = utiltesting.MakeClusterQueue("cq1").
+			ResourceGroup(
+				*utiltesting.MakeFlavorQuotas("f1").Resource(corev1.ResourceCPU, "4").Obj(),
+			).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+
+		lq = utiltesting.MakeLocalQueue("lq1", ns.Name).ClusterQueue("cq1").Obj()
+		gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
+
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor, true)
+	})
+
+	ginkgo.When("Kueue is started after import", func() {
+		ginkgo.It("Should keep the imported pods admitted", func() {
+
+			pod1 := utiltestingpod.MakePod("pod1", ns.Name).
+				Label("src.lbl", "src-val").
+				Request(corev1.ResourceCPU, "2").
+				Obj()
+			pod2 := utiltestingpod.MakePod("pod2", ns.Name).
+				Label("src.lbl", "src-val").
+				Request(corev1.ResourceCPU, "2").
+				Obj()
+
+			ginkgo.By("Creating the initial pods", func() {
+				gomega.Expect(k8sClient.Create(ctx, pod1)).To(gomega.Succeed())
+				gomega.Expect(k8sClient.Create(ctx, pod2)).To(gomega.Succeed())
+			})
+
+			ginkgo.By("Running the import", func() {
+				mapping, err := importerutil.LoadImportCache(ctx, k8sClient, []string{ns.Name}, "src.lbl", map[string]string{"src-val": "lq1"}, nil)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+				gomega.Expect(mapping).ToNot(gomega.BeNil())
+
+				gomega.Expect(importerpod.Check(ctx, k8sClient, mapping, 8)).To(gomega.Succeed())
+				gomega.Expect(importerpod.Import(ctx, k8sClient, mapping, 8)).To(gomega.Succeed())
+			})
+
+			wl1LookupKey := types.NamespacedName{Name: pod.GetWorkloadNameForPod(pod1.Name, pod1.UID), Namespace: ns.Name}
+			wl2LookupKey := types.NamespacedName{Name: pod.GetWorkloadNameForPod(pod2.Name, pod2.UID), Namespace: ns.Name}
+			wl1 := &kueue.Workload{}
+			wl2 := &kueue.Workload{}
+
+			ginkgo.By("Checking the Workloads are created and admitted", func() {
+				gomega.Expect(k8sClient.Get(ctx, wl1LookupKey, wl1)).To(gomega.Succeed())
+				gomega.Expect(k8sClient.Get(ctx, wl2LookupKey, wl2)).To(gomega.Succeed())
+
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1, wl2)
+			})
+
+			ginkgo.By("Starting kueue, the cluster queue status should account for the imported Workloads", func() {
+				fwk.StartManager(ctx, cfg, managerAndSchedulerSetup)
+
+				util.ExpectClusterQueueStatusMetric(cq, metrics.CQStatusActive)
+				gomega.Eventually(func(g gomega.Gomega) {
+					updatedQueue := &kueue.ClusterQueue{}
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), updatedQueue)).To(gomega.Succeed())
+					g.Expect(updatedQueue.Status.AdmittedWorkloads).To(gomega.Equal(int32(2)))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Comment:
At this point, let's also check that the workloads have the same UID (so they were not recreated)

Reply:
done